lib/entities/empty_state.rb

```diff
@@ -3,13 +3,22 @@ class Entities::Page::EmptyState
   def initialize(empty_state, parent_empty_state=nil)
     @heading = "You have no items."
     @message = "See <a href=\"https://www.lib.umich.edu/find-borrow-request\">what you can borrow from the library<\/a>."
+    @parent_empty_state = parent_empty_state
     
     if !empty_state.nil?
       @heading = empty_state["heading"] || @heading
       @message = empty_state["message"] || @message
     elsif !parent_empty_state.nil?
-      @heading = parent_empty_state.heading || @heading
-      @message = parent_empty_state.message || @message
+      @heading = @parent_empty_state.heading || @heading
+      @message = @parent_empty_state.message || @message
+    end
+  end
```

Comment:
Looking at this, the same should be done with `@heading` and `@message`

Make `empty_state` an instance variable and then use that in the `heading` and `message` methods rather than mutating `@heading` and `@message`.